src/ucs/config/parser.c

```diff
@@ -1420,6 +1421,51 @@ static ucs_status_t ucs_config_parser_get_sub_prefix(const char *env_prefix,
     return UCS_OK;
 }
 
+void ucs_config_parse_config_files()
+{
+    char config_files[5][MAXPATHLEN] = {{0}};
+    char *dlpath_dup = NULL, *ucx_config_dir = NULL;
+    int i;
+    Dl_info ucs_dl_info;
+
+    /* Global config dir */
+    strcpy(config_files[0], UCX_GLOBAL_CONFIG_FILE);
+
+    /* Library dir */
+    if (dladdr(ucs_config_parse_config_files, &ucs_dl_info) != 0) {
+        dlpath_dup = ucs_strdup(ucs_dl_info.dli_fname, "config_parser");
+        snprintf(config_files[1], MAXPATHLEN, "%s/../etc/%s",
+                 dirname(dlpath_dup), UCX_CONFIG_FILE_NAME);
+    }
+
+    /* User home dir */
+    snprintf(config_files[2], MAXPATHLEN, "%s/%s", getenv("HOME"),
+             UCX_CONFIG_FILE_NAME);
+
+    /* Custom conf dir */
```

Comment:
In additional to the initial 4 paths discussed, I added this one.
The logic is to allow the user to provide a custom path to his `ucx.conf` while which is located wherever he wants (the 4 other paths don't provide such option but only impose "our rules").